dd-java-agent/agent-iast/src/main/java/com/datadog/iast/taint/Ranges.java

```diff
@@ -319,4 +334,17 @@ static int calculateSubstringSkippedRanges(int offset, int length, @Nonnull Rang
 
     return skippedRanges;
   }
+
+  public static boolean areMarked(
+      @Nonnull final Range[] ranges, @Nonnull final VulnerabilityType type) {
+    if (ranges.length == 0) {
+      return false;
+    }
+    for (Range range : ranges) {
+      if ((range.getMarks() & type.mark()) < 1) {
```

Comment:
```suggestion
      if ((range.getMarks() & type.mark()) == 0) {
```

`1 << 31` is negative

Reply:
🤦‍♂️  done!